libs/util/str.spec.ts

```diff
@@ -154,3 +157,25 @@ test.each([
 ])('ipv6Regex fails: %s', (s) => {
   expect(IPV6_REGEX.test(s)).toBe(false)
 })
+
+test.each([
+  '',
+  '1',
+  'abc',
+  '256.1.1.1',
+  '2001:0db8:85a3:0000:0000:8a2e:0370:7334 ',
+  ' 2001:db8::',
+  '1:2:3:4:5:6:7:8:9',
+  '1:2:3:4:5:6::7:8',
+  ':1:2:3:4:5:6:7:8',
+  '1:2:3:4:5:6:7:8:',
+  '::1:2:3:4:5:6:7:8',
+  '1:2:3:4:5:6:7:8::',
+  '1:2:3:4:5:6:7:88888',
+  '2001:db8:3:4:5::192.0.2.33', // std::new::Ipv6Net allows this one
+  'fe08::7:8%',
+  'fe08::7:8i',
+  'fe08::7:8interface',
+])('validateIp catches invalid IP: %s', (s) => {
+  expect(validateIp(s)).toStrictEqual({ isv4: false, isv6: false, valid: false })
+})
```

Comment:
Since the tests for the regexes are so thorough, I think it would be ok to just do one or two for each of the possible outcomes here

```ts
test('validateIp', () => {
  expect(validateIp('').toEqual({ isv4: false, isv6: false, valid: false })
  expect(validateIp('1.1.1').toEqual({ isv4: false, isv6: false, valid: false })
  expect(validateIp('1.1.1.1').toEqual({ isv4: true, isv6: false, valid: true })
  // etc
})
```

Reply:
Oh, I didn't realize you had added a line to each of the other tests. Sorry!

In that case, do what you did, love it, and don't even export the regexes if we're not using them directly anywhere. Just test this function.